### DISCUSSION TITLE
How to define char type and its length for mysql?

### DISCUSSION THREAD

**Discussion by robincodex** [ent/ent#1664] [Q&A] [answered]
```go
// Fields of the User.
func (User) Fields() []ent.Field {
	return []ent.Field{
		field.String("password"),
	}
}
```
The password is an MD5 string and has a fixed length of 32, I want to use the char type, I think I should use the `field.Other` method, but I don't know what to do with the second parameter.

Update:
I customized the `Password` type by referring to the https://github.com/google/uuid/blob/master/sql.go, 
I called `Client.Schema.Create(ctx)` method to migrate the table and the first time it created it correctly, the second time there was an error `sql/schema: mysql: unknown char(32) type (not a uuid)`.
It seems that as long as the char type is considered to be UUID.
```go
type Password string

func (p Password) Value() (driver.Value, error) {
	return string(p), nil
}

func (p *Password) Scan(src interface{}) error {
	switch src := src.(type) {
	case nil:
		return nil

	case string:
		*p = Password(src)
		return nil

	case []byte:
		*p = Password(src)
		return nil

	default:
		return fmt.Errorf("Scan: unable to scan type %T into string", src)
	}
}

// Fields of the User.
func (User) Fields() []ent.Field {
	return []ent.Field{
		field.Other("password", Password("")).Sensitive().SchemaType(map[string]string{
			dialect.MySQL: "char(32)",
		})
	}
}
```

**Comment by a8m** [comment, marked as answer]
Hey @RobinCodeX,
I was able to restore your issue fixed this in https://github.com/ent/ent/pull/1668. After updating your version to `master`, you'll be able to use `field.Other` with custom `char` type (see example in the PR).

Regarding your implementation, may I ask why do you need a custom GoType for this field or `field.Other`? Why not to use the `field.String` option? You can configure it today in a few forms (see example in the PR):

```go
// Add a Go validator for this field (length is <= 32),
// and sets the schema-type to `varchar(32)`.
field.String("password").
	Sensitive().
	Size(32)

// Add a string field (as above), with all its predicates,
// but configure a custom dialect-specific type.
field.String("password").
	Sensitive().
	SchemaType(map[string]string{
		dialect.MySQL: "char(32)",
	})

// Same example as above, but use a custom Go type.
field.String("password").
	Sensitive().
	GoType(Password("example"))
	SchemaType(map[string]string{
		dialect.MySQL: "char(32)",
	})
```

**Reply by robincodex** [reply]
Thank you, here is the code I used for the first time when there was an error `sql/schema: mysql: unknown char(32) type (not a uuid)`, so I guess I might need `field.Other`
```go
field.String("password").
	Sensitive().
	SchemaType(map[string]string{
		dialect.MySQL: "char(32)",
	})
```
